opal/mca/btl/base/btl_base_am_rdma.c

```diff
@@ -794,6 +799,65 @@ static int mca_btl_base_am_rdma_progress (void)
     }));
 }
 
+static int mca_btl_base_am_atomic_lock_64 (int64_t *operand, opal_atomic_int64_t *addr,
+                                           mca_btl_base_atomic_op_t op, opal_mutex_t *lock) {
+
+    int64_t result = 0;
+
+    if(lock) {
+        OPAL_THREAD_LOCK(lock);
+    }
+
+    switch (op) {
+    case MCA_BTL_ATOMIC_ADD:
+        result = *addr;
+        *addr += *operand;
+        break;
+    case MCA_BTL_ATOMIC_AND:
+        result = *addr;
+        *addr &= *operand;
+        break;
+    case MCA_BTL_ATOMIC_OR:
+        result = *addr;
+        *addr |= *operand;
+        break;
+    case MCA_BTL_ATOMIC_XOR:
+        result = *addr;
+        *addr ^= *operand;
+        break;
+    case MCA_BTL_ATOMIC_SWAP:
+        result = *addr;
+        *addr  = *operand;
+        break;
+    case MCA_BTL_ATOMIC_MIN:
+        result = *addr;
+        if(result > *operand) {
+            *addr = *operand;
+        }
+        break;
+    case MCA_BTL_ATOMIC_MAX:
+        result = *addr;
+        if(result < *operand) {
+            *addr = *operand;
+        }
+        break;
+    default: {
+        if(lock) {
```

Comment:
This is the internal version of the op, I am almost certain we can't hit this code with an incorrect opcode (all the checks should have been done at the upper level).